File: cdk-from-cfn-testing/expected/condition_sub/csharp/ConditionSubConstruct.cs
using Amazon.CDK;
using Amazon.CDK.AWS.SNS;
using Constructs;
using System.Collections.Generic;

namespace ConditionSubConstruct
{
    public class ConditionSubConstructProps
    {
        public string EnvName { get; set; }

    }

    /// <summary>
    /// Exercises Fn::Sub inside a Condition
    /// </summary>
    public class ConditionSubConstruct : Construct
    {
        public ConditionSubConstruct(Construct scope, string id, ConditionSubConstructProps props = null) : base(scope, id)
        {

            // Conditions
            bool hasCustomSuffix = $"{props.EnvName}-suffix" == "prod-suffix";

            // Resources
            var topic = new CfnTopic(this, "Topic", new CfnTopicProps
            {
            });
        }
    }
}

File: cdk-from-cfn-testing/expected/condition_sub/typescript/ConditionSubConstruct.ts
import * as cdk from 'aws-cdk-lib';
import * as sns from 'aws-cdk-lib/aws-sns';
import { Construct } from 'constructs';

export interface ConditionSubConstructProps {
  /**
   */
  readonly envName: string;
}

/**
 * Exercises Fn::Sub inside a Condition
 */
export class ConditionSubConstruct extends Construct {
  public constructor(scope: Construct, id: string, props: ConditionSubConstructProps) {
    super(scope, id);

    // Conditions
    const hasCustomSuffix = `${props.envName!}-suffix` === 'prod-suffix';

    // Resources
    const topic = hasCustomSuffix
      ? new sns.CfnTopic(this, 'Topic', {
        })
      : undefined;
    if (topic != null) {
    }
  }
}
